lib/crystalball/map_generator/helpers/path_filter.rb

```diff
@@ -18,6 +23,8 @@ def filter(paths)
           paths
             .select { |file_name| file_name.start_with?(root_path) }
             .map { |file_name| file_name.sub("#{root_path}/", '') }
+            .reject { |file_name| exclude_sources.any? do |pattern| pattern.match?(file_name) end }
```

Comment:
This is ugly right now, I'll refactor into a method and make it type safe for both strings and regexp instances and collections.